src/main/java/io/cdap/e2e/pages/actions/CdfPipelineRunAction.java

```diff
@@ -107,10 +117,27 @@ public static void waitForPipelineToTransitionToStatus(String pipelineStatus, lo
 
   /**
    * Wait till the Pipeline's status changes (from Running) to either Succeeded, Failed or Stopped within the
-   * Timeout: {@link ConstantsUtil#PIPELINE_RUN_TIMEOUT_SECONDS}
+   * Timeout: {@link ConstantsUtil#IMPLICIT_TIMEOUT_SECONDS}
    */
-  public static void waitTillPipelineRunCompletes() {
-    SeleniumDriver.getWaitDriver(ConstantsUtil.PIPELINE_RUN_TIMEOUT_SECONDS).until(ExpectedConditions.or(
+  public static void waitTillPipelineRunCompletes() throws InterruptedException, IOException {
+    int pipelineExecutionTimeFlag = 0;
+    // Adding a page refresh in case tests are running on CDF to update the pipeline status.
+    if (Boolean.parseBoolean(SeleniumHelper.readParameters(ConstantsUtil.TESTONCDF)) ||
+      Boolean.parseBoolean(SeleniumHelper.readParameters(ConstantsUtil.TESTONHDF))) {
+      // Adding pipelineExecutionTimeFlag to break the loop if pipeline status is Running state for more than
+      // 900 seconds.
+      do {
+        pipelineExecutionTimeFlag += 120;
```

Comment:
Can we please define a retry function which takes a retry delay, max retry delay and max retrycount. This can be used both here and in clickIfDisplayed

Reply:
Also please make the retry delays as constants

Reply:
Added the retry function and re-used it for both pipeline refresh and button click. (5c54894de6f63366c43ab25cc9655e52f9c1d468)